2nd attempt at adding json exporting!

ArchooD2 opened the pull request:
Hope this looks right, still haven't added tests in the suite.

Commit:
feat: add export functionality for query tracking logs

Commit:
lint: `black .` command ran

MuddyHope commented:
@ArchooD2, can you add a docstring or suppress the lint issue ? Also some test cases, once you are done.

Commit:
docs: add export_log method docstring to clarify parameters (pylint...)

ArchooD2 commented:
adding test cases

MuddyHope commented:
There is another conflict. Will test it once you are done.

Commit:
fix: clean up docstring formatting in export_log method and add test for JSON log export

ArchooD2 commented:
Cool, added test case and reformatted with black and pylint testing 👍 Let me know if anything needs changing

MuddyHope commented:
There are 2 files that have conflicts, can you resolve them? @ArchooD2

ArchooD2 committed:
Merge branch 'main' into json-exporter-v2

ArchooD2 commented:
Is that fully fixed?

MuddyHope commented on tests/test_core.py:
```diff
@@ -68,3 +70,28 @@ def do_work(self, a, b):
     assert "ms" in record.message
+
+
+def test_log_export_to_json(tmp_path, monkeypatch):
```
Can you move this test function into a new directory new file, **exporter** ?

ArchooD2 replied:
Done in commit 918a44e

MuddyHope commented on tests/test_core.py:
```diff
@@ -68,3 +70,28 @@ def do_work(self, a, b):
+    export_path = tmp_path / "log.json"
+
+    # Monkeypatch the config to enable JSON export
+    config = get_config()
```
We won't be using get_config, we will be using **configure**, look at [examples/core/with_config.py](url), can you modify the config.py also?

MuddyHope commented on pyquerytracker/core.py:
```diff
@@ -1,8 +1,10 @@
 import time
 import logging
+import json
+import os
```
Use pathlib, instead of os

ArchooD2 replied:
Fixed in commit 918a44e!

MuddyHope commented on pyquerytracker/core.py:
```diff
@@ -38,65 +41,90 @@ def my_function():
+        Parameters:
+            log_data (dict): The data to log and export.
+        """
+        if self.config.export_path and self.config.export_type == ExportType.JSON:
```
Make File write thread-safe

ArchooD2 replied:
Done in push 918a44e

MuddyHope commented on pyquerytracker/core.py:
```diff
@@ -38,65 +41,90 @@ def my_function():
+                        extra=log_data,
                     )
+
+                self.export_log(log_data)
```
Refactor JSON Export Logic into Reusable Class

ArchooD2 replied:
Attempted in commit 918a44e. Let me know next steps

MuddyHope commented on pyquerytracker/core.py:
```diff
@@ -38,65 +41,90 @@ def my_function():
+        """
+        if self.config.export_path and self.config.export_type == ExportType.JSON:
+            os.makedirs(os.path.dirname(self.config.export_path), exist_ok=True)
+            with open(self.config.export_path, "a", encoding="utf-8") as f:
```
Currently, writing to os.path.dirname(None) will crash if export_path isn’t set. You should:

Skip export entirely if export_path is None

ArchooD2 replied:
Fixed in commit 918a44e, i believe

Commit:
Moved json exporting to separate file with class,
Moved json export testing to new file
Changed get_config patch in test to configure() call
No longer writes to os.path.dirname(None) (disables JsonExporter class entirely if None path supplied)
File writes should now be thread-safe
Changed export logic to use pathlib rather than os

MuddyHope committed:
Examples test runs for Export added

MuddyHope commented on tests/exporter/test_json_exporter.py:
```diff
@@ -0,0 +1,24 @@
+
+
+def test_log_export_to_json(tmp_path):
+    log_file = tmp_path / "out.jsonl"
```
Can you make it json, instead of jsonl, we can take this up next improvement.
I have added an example too. Thank you.

MuddyHope replied:
Also preferably a log stating where the file is stored.

Commit:
fix: update log file extension to .json and add debug logging for export (requested changes)

MuddyHope commented:
Thanks @ArchooD2, for you patience.

MuddyHope commented:
Can you cherry-pick these commits by forking the repo and contributing?

<img width="974" alt="image" src="https://github.com/user-attachments/assets/cbd3e359-ef14-4b7c-8cd7-d8c00f88549e" />

ArchooD2 commented:
https://github.com/MuddyHope/pyquerytracker/pull/36

MuddyHope commented:
@ArchooD2, closing this PR.
